lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -133,20 +132,41 @@ actor ConnectorSink is Sink
   var _reconnect_pause: U64
   var _host: String
   var _service: String
+  let _worker_name: WorkerName
+  let _protocol_version: String
+  let _cookie: String
+  let _auth: ApplyReleaseBackpressureAuth
   var _from: String
 
   // Producer (Resilience)
   let _timers: Timers = Timers
 
   var _seq_id: SeqId = 0
 
+  // 2PC
```

Comment:
Can we encapsulate all of this 2PC related stuff in a class?

Reply:
I second that motion. It seems that a lot of the 2PC work this module is doing could be delegated to a dedicated, content-agnostic class for doing just that.
It would also make it much easier to test in isolation.